practices/multiple_environments.md

```diff
@@ -0,0 +1,125 @@
+---
+slug: multiple-environments
+stages:
+  - development
+short_description: Multi-environment usage is a recommended practice for software development as it helps in maintaining an optimal workflow for the project which overall helps in delivering the project in a reliable and timely manner.  
+tags: 
+  - continuous integration
+  - continuous delivery
+  - software development
+  - quality control
+  - development environment
+  - testing environment
+  - staging environment
+  - production environment 
+  - performance testing
+  - continuous integration
+  - revision control
+keywords:
+  - continuous integration
+  - continuous delivery
+  - effective deployment
+  - load testing
+  - devops
+  - git
+  - testing
+  - testing environment
+  - staging environment
+  - production environment
+  - continuous integration
+---
+
+# Multiple Environments
+
+**TL;DR**
+
+Multi-environment usage is a recommended practice for software development as it helps in maintaining an optimal workflow for the project which overall helps in delivering the project in a reliable and timely manner. This structured release management process allows rollout, testing, and rollback in case of problems.
+
+## What Is a Multi Environments
+
+Multi-environment is a recommended practice that allows you to ship code to the end-user without any bugs or issues. This type of practice is very reliable as it involves multiple stages or environments. It also helps the company to reply to the user's feedback efficiently and immediately.
+
+These environments are divided according to the tiers of the project. A particular individual related to a specific environment may not have access to other environments to avoid security breaches or human errors.
+
+We can classify Multi-environments into different stages, according to the level of development across environments:
+
+1. **Development environment**
+   The development environment (dev) is the environment in which software changes are developed, most simply an individual developer's workstation. This environment includes development tools like a compiler, integrated development environment, different or additional versions of libraries and support software, etc., which are not present in a user's environment.
```

Comment:
Please, rewrite this using your own words. It is too similar to the Wiki article: _The development environment (dev) is the environment in which changes to software are developed, most simply an individual developer's workstation. This (..) environment will include development tools like a compiler, integrated development environment, different or additional versions of libraries and support software, etc., which are not present in a user's environment._

Reply:
Alternatively, the Henricodofin article has text with similarities: _The development environment is the environment in which changes to software are developed, most simply an individual developer's workstation. This differs from the ultimate target environment in various ways – the target may not be a desktop computer (it may be a smartphone, embedded system, headless machine in a data center, etc.), and even if otherwise similar, the developer's environment will include development tools like a compiler, integrated development environment, different or additional versions of libraries and support software, etc., which are not present in a user's environment._